lib/index.js

```diff
@@ -33,7 +33,8 @@ module.exports = function rewireify(file, options) {
 
   function end() {
     post += "/* This code was injected by Rewireify */\n";
-    post += "if (Object.isExtensible(module.exports)) {\n";
+    post += "if ('functionobject'.match(typeof module.exports) && \n"
```

Comment:
You're right, it's much more readable that way. I'll fix that now. 

Also, as this is just to stop exceptions from calling `Object.isExtensible` on primitives, I was meaning the fact that it would match arrays as a positive! (And not a false one ;) )

Sent from my phone

> El 18/11/2015, a las 10:37, Matt Hinchliffe <notifications@github.com> escribió:
> 
> In lib/index.js:
> 
> > @@ -33,7 +33,8 @@ module.exports = function rewireify(file, options) {
> >  
> >    function end() {
> >      post += "/* This code was injected by Rewireify */\n";
> > -    post += "if (Object.isExtensible(module.exports)) {\n";
> > +    post += "if ('functionobject'.match(typeof module.exports) && \n"
> Apologies, I interpreted it the wrong way around. I'll blame it on my brain is still warming up after holidays.
> 
> This is totally fine but readability should always trump terseness. To me it reads more naturally in the reverse order:
> 
> (typeof module.exports).match(/object|function/)
> I'm not worried about any false positives such as matching an array, if that happened to be the case then the end user is may have larger problems to resolve.
> 
> —
> Reply to this email directly or view it on GitHub.
>